Sources/Metrics/Metrics.swift

```diff
@@ -74,3 +74,22 @@ extension Timer {
         }
     }
 }
+
+#if swift(>=5.7)
+extension Timer {
+    /// Convenience for recording a duration based on ``Duration``.
+    /// Duration will be recorded in nanoseconds.
+    ///
+    /// - parameters:
+    ///     - duration: The duration to record.
+    @available(macOS 13, iOS 16, tvOS 15, watchOS 8, *)
+    @inlinable
+    public func record(_ duration: Duration) {
+        // `Duration` doesn't have a nice way to convert it nanoseconds or seconds,
+        // so we'll do the multiplication manually.
+        // nanoseconds are the smallest unit Timer can track, so we'll record in that.
+        let durationNanoseconds = duration.components.seconds * 1_000_000_000 + duration.components.attoseconds / 1_000_000_000
```

Comment:
can `duration.components.attoseconds ` ever be zero?

Reply:
- **can duration.components.attoseconds ever be zero?** 
  Unlikely, but they can. That shouldn't be a problem, though? I think it's `Int64` divided by `Int64` that is not zero, so it'll just be zero? 
- **should we be using `multipliedReportingOverflow` and `dividedReportingOverflow` to address overflow risk?**
  TIL! I was suspicious about overflows, thank you for pointing out the direction. I'll take a look and use those, happy to address in this or a new PR.

Reply:
>  I'll take a look and use those, happy to address in this or a new PR.

lets add it here to make sure the code is safe

Reply:
@tomerd, here's an attempt: https://github.com/apple/swift-metrics/pull/133/files#diff-19a0ef33862d9079451ca7ba346b734175c62c5ad95fb30d565a41f9ea32a770R95 

- - `dividedReportingOverflow` should not be applicable in here, since we know we're not dividing by zero. 
- `multipliedReportingOverflow` and `addingReportingOverflow` will return a `Bool` flag on error. Not a very common way of error handling in Swift, but that's cheaper than making a new `Error` and throwing. So, I considered keeping a similar API, and perhaps providing a convenience func that would be throwing, but decided `throws` provides a more familiar and simple API.

What I don't know is whether there's a performance hit we take for marking the function as throwing, in situations where we are not actually throwing errors. I _think_ it's negligible?